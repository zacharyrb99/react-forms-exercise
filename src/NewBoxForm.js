import {useState} from 'react';

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        width: '',
        height: '',
        color: ''
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({...formData});
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor='width'>Width: </label>
            <input id='width' type='number' name='width' placeholder='Box Width' value={formData.width} onChange={handleChange} />

            <label htmlFor='height'>Height: </label>
            <input id='height' type='number' name='height' placeholder='Box Height' value={formData.height} onChange={handleChange} />

            <label htmlFor='color'>Color: </label>
            <input id='color' type='color' name='color' placeholder='Box Color' value={formData.color} onChange={handleChange} />

            <button>Add Box!</button>
        </form>
    )
}

export default NewBoxForm;