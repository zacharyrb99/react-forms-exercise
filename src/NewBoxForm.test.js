import {render} from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('NewBoxForm renders without crashing', () => {
    render(<NewBoxForm/>);
});

test('NewBoxForm matches snapshot', () => {
    const {asFragment} = render(<NewBoxForm/>);
    expect(asFragment).toMatchSnapshot();
});