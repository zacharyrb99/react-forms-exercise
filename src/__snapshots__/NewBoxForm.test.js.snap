// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm matches snapshot 1`] = `[Function]`;
